src/main/java/tech/jhipster/lite/module/infrastructure/secondary/javadependency/gradle/GradleCommandHandler.java

```diff
@@ -269,4 +283,99 @@ private void addPluginConfiguration(GradlePluginConfiguration pluginConfiguratio
       ContentReplacers.of(new MandatoryFileReplacer(new JHipsterProjectFilePath(BUILD_GRADLE_FILE), replacer))
     );
   }
+
+  private void addProfileIdentifier() {
+    MandatoryReplacer replacer = new MandatoryReplacer(
+      new RegexNeedleBeforeReplacer(
+        (contentBeforeReplacement, newText) -> !contentBeforeReplacement.contains(newText),
+        GRADLE_PROFILE_ACTIVATION_NEEDLE
+      ),
+      """
+      val profiles = (project.findProperty("profiles") as String? ?: "")
+        .split(",")
+        .map { it.trim() }
+        .filter { it.isNotEmpty() }\
+      """
+    );
+    fileReplacer.handle(
+      projectFolder,
+      ContentReplacers.of(new MandatoryFileReplacer(new JHipsterProjectFilePath(BUILD_GRADLE_FILE), replacer))
+    );
+  }
+
+  private void addProcessResourcesApplicationReplacer() {
+    MandatoryReplacer replacer = new MandatoryReplacer(
+      new RegexNeedleBeforeReplacer(
+        (contentBeforeReplacement, newText) -> !contentBeforeReplacement.contains(newText),
+        GRADLE_PROCESS_RESOURCES_NEEDLE
+      ),
+      """
+        filesMatching("**/application.yml") {
+          filter {
+            // jhipster-needle-gradle-profiles-properties
+          }
+        }\
+      """
+    );
+    fileReplacer.handle(
+      projectFolder,
+      ContentReplacers.of(new MandatoryFileReplacer(new JHipsterProjectFilePath(BUILD_GRADLE_FILE), replacer))
+    );
+  }
+
+  private void enablePrecompiledScriptPlugins() {
+    addFileToProject("buildtool/gradle/buildSrc/build.gradle.kts.template", "buildSrc/build.gradle.kts");
+  }
+
+  private void addProfile(AddJavaBuildProfile command) {
+    addProfileActivation(command);
+    addProfileBuildGradleFile(command);
+  }
+
+  private void addProfileActivation(AddJavaBuildProfile command) {
+    String profileTemplate =
+      """
+      if (profiles.contains("%s")) {
+        apply(plugin = "profile-%s")
+      }\
+      """;
```

Comment:
There's no need for the `profileTemplateFilled` variable. Just do:
```suggestion
      """.formatted(command.buildProfileId(), command.buildProfileId());
```